app/api/dao/user.py

```diff
@@ -143,27 +146,30 @@ def get_user_by_username(username: str):
         return UserModel.find_by_username(username)
 
     @staticmethod
-    def list_users(user_id: int, search_query: str = "", is_verified = None):
+    def list_users(user_id: int, search_query: str = "", page: int = DEFAULT_PAGE, per_page: int = DEFAULT_USERS_PER_PAGE, is_verified = None):
         """ Retrieves a list of verified users with the specified ID.
         
         Arguments:
             user_id: The ID of the user to be listed.
             search_query: The search query for name of the users to be found.
             is_verified: Status of the user's verification; None when provided as an argument.
+            page: The page of users to be returned
+            per_page: The number of users to return per page
         
         Returns:
             A list of users matching conditions and the HTTP response code.
         
         """
 
-        users_list = UserModel.query.filter(UserModel.id != user_id).all()
+        users_list = UserModel.query.filter(
+            UserModel.id != user_id,
+            not is_verified or UserModel.is_email_verified,
+            func.lower(UserModel.name).contains(search_query.lower())
+        ).paginate(page=page, per_page=per_page, error_out=False, max_per_page=UserDAO.DEFAULT_USERS_PER_PAGE).items
```

Comment:
The `max_per_page` does not allow me to set `per_page` of 20, making `max_per_page` set to UserDAO.DEFAULT_USERS_PER_PAGE which is 10, sounds a bit limiting. What if we have a UserDAO.MAX_USERS_PER_PAGE? or we don't set `max_per_page` at all 🤔 

My request:
`curl -X GET "http://localhost:5000/users?page=1&per_page=20" -H "accept: application/json" -H "Authorization: My Token"`

The result was 10 items.

Reply:
@isabelcosta right, we set `max_per_page` to 10 in a previous conversation. I agree we can increase that with a different constant or leave it out all together and not have a `max_per_page`. I think it would be wise to have a `max_per_page` set in the case that someone requests an enormous number. thoughts? I'll make a `UserDAO.MAX_USERS_PER_PAGE` constant and set it to 50 for now and you can let me know if you want something different :)

Reply:
I totally agree with your suggestion :) @lmcicat